### PR TITLE
ipv6: add support for ipv6 addresses lists

### PR DESCRIPTION
**Pull request by IvanNardi**

### CODE REVIEW

File: example/ndpiReader.c [hunk 1]
```diff
@@ -3901,6 +3901,9 @@ static void printResults(u_int64_t processing_time_usec, u_int64_t setup_time_us
       printf("\tPatricia protocols:   %llu/%llu (search/found)\n",
 	     (long long unsigned int)cumulative_stats.patricia_stats[NDPI_PTREE_PROTOCOLS].n_search,
 	     (long long unsigned int)cumulative_stats.patricia_stats[NDPI_PTREE_PROTOCOLS].n_found);
+      printf("\tPatricia protocols 6: %llu/%llu (search/found)\n",
```
Comment (utoni) [inline example/ndpiReader.c]: What about something like:

```suggestion
      printf("\tPatricia protocols IPv6: %llu/%llu (search/found)\n",
```

File: src/lib/ndpi_main.c [hunk 2]
```diff
@@ -2475,6 +2479,52 @@ u_int16_t ndpi_network_port_ptree_match(struct ndpi_detection_module_struct *ndp
...
+					 struct in6_addr *pin,
+					 u_int16_t port /* network byte order */)
+{
+  ndpi_prefix_t prefix;
+  ndpi_patricia_node_t *node;
+
+  if(!ndpi_str || !ndpi_str->protocols_ptree6)
+    return(NDPI_PROTOCOL_UNKNOWN);
+
+  /* TODO: check on "private" addresses? */
```
Comment (utoni) [inline src/lib/ndpi_main.c:2492]: Makes sense for me. There could be also risk set, if that happens.

Comment (IvanNardi) [reply src/lib/ndpi_main.c:2492]: More than an anomaly, it is an optimization. Like in the ipv4 case, if there are not link local/"private" addresses in the tree (and, by default, there are not) we can avoid the lookup if the flow addressees are of these types

File: example/ndpiReader.c (hunk 1 above)
Comment (IvanNardi) [reply example/ndpiReader.c]: Done

**Review by utoni — APPROVED** [review]

**Pull request merged**